src/hooks/userspace/hook-grpc.js

```diff
@@ -160,58 +160,38 @@ function sendMetadataWrapper(rootContext) {
  * @param {string} requestName The human-friendly name of the request.
  */
 function wrapUnary(namespace, handlerSet, requestName) {
-  var rootContext;
-  // CLS context object. Because a root span is started outside of the unary function
-  // handle, we use this object to explicitly ensure that any client trace spans
-  // have the same context available to them.
-  var context = {};
-  // Start right before deserialization
-  // End right after serialization
-  shimmer.wrap(handlerSet, 'deserialize', function (deserialize) {
-    return namespace.bind(function deserializeTrace() {
-      rootContext = startRootSpanForRequest(requestName);
-      return deserialize.apply(this, arguments);
-    }, context);
-  });
-  // Even though our root span doesn't start or end here,
-  // we need to wrap 'func' so that:
-  // (1) child spans have the correct context, and
-  // (2) we can get additional labels as needed
   shimmer.wrap(handlerSet, 'func', function (func) {
-    return namespace.bind(function funcTrace(call, callback) {
-      if (agent.config_.enhancedDatabaseReporting) {
-        shimmer.wrap(call, 'sendMetadata', sendMetadataWrapper(rootContext));
-      }
-      if (agent.config_.enhancedDatabaseReporting) {
-        rootContext.addLabel('argument', JSON.stringify(call.request));
-      }
-      // arguments[1] is the callback
-      arguments[1] = function (err, result, trailer, flags) {
+    return function funcTrace(call, callback) {
+      var that = this;
+      var args = arguments;
+      // Running in the namespace here propagates context to func.
+      return namespace.runAndReturn(function() {
+        var rootContext = startRootSpanForRequest(requestName);
         if (agent.config_.enhancedDatabaseReporting) {
-          if (err) {
-            rootContext.addLabel('error', err);
-          } else {
-            rootContext.addLabel('result', JSON.stringify(result));
-          }
-          if (trailer) {
-            rootContext.addLabel('trailing_metadata', JSON.stringify(trailer.getMap()));
-          }
+          shimmer.wrap(call, 'sendMetadata', sendMetadataWrapper(rootContext));
```

Comment:
It's per request; specifically, it's passed into the request handler and contains information about the request (including the incoming request data, namely `call.request`).